cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	calculationsv1 "github.com/vega-project/ccb-operator/pkg/apis/calculations/v1"
+	"github.com/vega-project/ccb-operator/pkg/client/clientset/versioned/fake"
+	v1 "github.com/vega-project/ccb-operator/pkg/client/clientset/versioned/typed/calculations/v1"
+	"github.com/vega-project/ccb-operator/pkg/util"
+)
+
+func TestCreateCalculation(t *testing.T) {
+	fakecs := fake.NewSimpleClientset()
+	var fakeClient v1.VegaV1Interface
+	fakeClient = fakecs.VegaV1()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	//teff := 10000
+	// testCalculationCount := teff + 2
+	// var dryCalcList []*calculationsv1.Calculation
+	// for teff != testCalculationCount {
+	// 	teff++
+	// 	newCalc := util.NewCalculation(float64(teff), 4.0)
+	// 	newCalc.DBKey = fmt.Sprintf("vz.star:teff_%d", teff)
+
+	// 	dryCalcList = append(dryCalcList, newCalc)
+	// 	logrus.WithFields(logrus.Fields{"calculation": newCalc.Name, "teff": teff, "logG": "4.0"}).Info("Creating calculation for test purpose")
+	// 	_, err := fakeClient.Calculations().Create(ctx, newCalc, metav1.CreateOptions{})
+	// 	if err != nil {
+	// 		t.Errorf("Coudn't create the calculation: %v", newCalc)
+	// 	}
+	// }
+
+	s := server{
+		logger: logrus.WithField("test-name", "get calculations test"),
+		ctx:    ctx,
+		client: fakeClient,
+	}
+
+	var jsonStr = []byte(`{"teff": "10000.0", "logG": "4"}`)
+	req, err := http.NewRequest("POST", "/calculations/create", bytes.NewBuffer(jsonStr))
+	if err != nil {
+		t.Fatal(err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(s.createCalculation)
+	handler.ServeHTTP(rr, req)
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("handler returned wrong status code: got %v want %v",
+			status, http.StatusOK)
+	}
+	expected := `{"teff": "10000.0", "logG": "4"}`
```

Comment:
This is not what we expect here. If the request will be successful we expect to get a calculation object in JSON format.
Also, please check different cases as well.

Reply:
Also, since we have a fake client here, I would like to have a check that the calculation was actually created.